File: 206-reverse-linked-list/206-reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode *prev,*temp;
        prev=NULL;
        temp=NULL;
        while(head!=NULL)
        {
            temp=head->next;
            head->next=prev;
            prev=head;
            head=temp;
        }
        return prev;
    }
};

File: 206-reverse-linked-list/206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        Stack<Integer> stack = new Stack();
        while(head != null) {
            stack.push(head.val);
            head = head.next;
        }
        ListNode newNode = null;
        ListNode newHead = null;
        while (stack.size() > 0) {
            if(newHead == null) {
                newNode = new ListNode(stack.pop());
                newHead = newNode;
                
            } else {
                newNode.next = new ListNode(stack.pop());
                newNode = newNode.next;
            }
        }
        return newHead;
    }
}